### PR TITLE
Add Basic use of a template

### PR DESCRIPTION
**Pull request by joshxg**
Basic example of template usage!

### CODE REVIEW

File: backend/pkg/scheduler.go [hunk 1]
```diff
@@ -118,18 +119,57 @@ func createReports() {
 
 	qr, err := api.QueryFromResponse(response)
 
-	f, e := excelize.OpenFile("./data/test.xlsx")
+	// Open the template
+	f, e := excelize.OpenFile("./data/template.xlsx")
+
+	// Regexp which matches a cell reference [TODO: Handle $ refs]
+	reg := regexp.MustCompile("([A-Za-z]+)|([0-9]+)")
```
Comment (wlthomson) [inline backend/pkg/scheduler.go:126]: Maybe missing something, but should the `|` be here? E.g. should a cell be `A3` (rather than `A` or `3`)?

File: backend/pkg/scheduler.go [hunk 2]
```diff
@@ -118,18 +119,57 @@ func createReports() {
...
 
+	// Replace the header row with the column text values.
 	columns := []string{"A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M"}
-	ri := 0
+	ci := 0
+	for _, column := range qr.Results.A.Tables[0].Columns {
+		cellRef := columns[ci] + strconv.Itoa(headerIndex)
+		f.SetCellValue("Sheet1", cellRef, column.Text)
+		ci += 1
+	}
```
Comment (wlthomson) [inline backend/pkg/scheduler.go:157]: https://github.com/openmsupply/msupply-dashboard-app/pull/13#discussion_r519489793 might also be useful here?

File: backend/pkg/scheduler.go [hunk 3]
```diff
@@ -118,18 +119,57 @@ func createReports() {
...
+	f.SetCellValue("Sheet1", result[0], "This is a dynamically added title")
+
+	// Same for the date
+	result = f.SearchSheet("Sheet1", "{{date}}")
+	f.SetCellValue("Sheet1", result[0], "This is a dynamically added date")
+
+	// Find where to start the rows from
+	rowStart := f.SearchSheet("Sheet1", "{{rows}}")
+	split := reg.FindAllString(rowStart[0], 2)
+	rowIndex, _ := strconv.Atoi(split[1])
```
Comment (wlthomson) [inline backend/pkg/scheduler.go:139]: Does ``{{rows}}`` have a special meaning here, or something hardcoded into the excel doc?

File: backend/pkg/scheduler.go [hunk 4]
```diff
@@ -118,18 +119,57 @@ func createReports() {
...
+		ci += 1
+	}
+
+	// Duplicate enough rows
+	ri := rowIndex + 1
+	i := 0
+	for i < len(qr.Results.A.Tables[0].Rows)-1 {
+		f.DuplicateRow("Sheet1", rowIndex)
+		i += 1
+	}
```
Comment (wlthomson) [inline backend/pkg/scheduler.go:165]: Can I get an ELI5 on this bit?

**Review by wlthomson — APPROVED** [review]
lgtm

File: backend/pkg/scheduler.go (hunk 3 above)
Comment (joshxg) [reply backend/pkg/scheduler.go:139]: just a symbol to be put into templates to signal where rows should go

File: backend/pkg/scheduler.go (hunk 4 above)
Comment (joshxg) [reply backend/pkg/scheduler.go:165]: Basically seemed like the easiest way of copying a row from the template and applying whatever style it has to the injected row

So, you make a template and you give some styles to a row - background colour etc. Then you need to apply that to each of the rows in the table, but the number of rows varies
